driver: sensor: adxl367: Fix for conversion and SPI

vladislav-pejic opened the pull request:
This is a fix for:
- Extra 0 in conversion of acceleration values in adxl367_accel_convert.
- Conversion of temperature values in adxl367_temp_convert.
- SPI communication when RTIO is used in SPI driver. When in adxl367_bus_access, const struct spi_buf buf[3] is used, spi_rtio_copy function won't set correct buffers and length for buf[2] and that will cause exception when that buffer is processed.

Signed-off-by: Vladislav Pejic [vladislav.pejic@orioninc.com](mailto:vladislav.pejic@orioninc.com)

MaureenHelm reviewed (commented):
Since these are three separate bug fixes, can you please split them into three separate commits (they can all stay in this PR)? This will help if we later find that one of the fixes was incorrect and can revert just that part.

vladislav-pejic committed:
driver: sensor: adxl367: Fix for SPI

Fix for SPI communication when RTIO is used in SPI driver. When in
adxl367_bus_access, const struct spi_buf buf[3] is used,
spi_rtio_copy function won't set correct buffers and length
for buf[2] and that will cause exception when that buffer is
processed.

Signed-off-by: Vladislav Pejic <vladislav.pejic@orioninc.com>

vladislav-pejic committed:
driver: sensor: adxl367: Temp conversion

Fix for conversion of temperature values in adxl367_temp_convert
and missing break statement.

Signed-off-by: Vladislav Pejic <vladislav.pejic@orioninc.com>

vladislav-pejic committed:
driver: sensor: adxl367: Fix for extra 0

Fix for extra 0 in conversion of acceleration values in
adxl367_accel_convert.

Signed-off-by: Vladislav Pejic <vladislav.pejic@orioninc.com>

vladislav-pejic commented:
> Since these are three separate bug fixes, can you please split them into three separate commits (they can all stay in this PR)? This will help if we later find that one of the fixes was incorrect and can revert just that part.

Done. Is it ok like this?
